pkg/utils/kubernetes/serviceaccount.go

```
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package kubernetes
+
+import (
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+)
+
+// Types extracted and adapted from kubernetes/kubernetes:
+// https://github.com/kubernetes/kubernetes/blob/b4980b8d6e65602818ad27eab6c9c12398569420/pkg/serviceaccount/claims.go#L52-L68
+
+type privateClaims struct {
+	Kubernetes kubernetes `json:"kubernetes.io,omitempty"`
+}
+
+type kubernetes struct {
+	Namespace string `json:"namespace,omitempty"`
+	Svcacct   ref    `json:"serviceaccount,omitempty"`
+}
+
+type ref struct {
+	Name string `json:"name,omitempty"`
+	UID  string `json:"uid,omitempty"`
+}
+
+// ExtractServiceAccountUID extracts the UID of the service account from the given Kubernetes JWT token string.
+// It does not verify the token signature and should only be used in trusted environments.
+func ExtractServiceAccountUID(tokenStr string) string {
+	parsed, err := jwt.ParseSigned(tokenStr, []jose.SignatureAlgorithm{jose.HS256})
+	if err != nil {
+		return ""
+	}
+	var claims privateClaims
+	if err := parsed.UnsafeClaimsWithoutVerification(&claims); err != nil {
+		return ""
```

Why are we ignoring errors here?

---

Returning errors [now](https://github.com/gardener/gardener/pull/13630/commits/58cfa9e324b830641ea56081783d82dc2c4bfcf9)